Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs

```diff
@@ -244,63 +220,50 @@ private void DeclareFeatureMessagesFactoryMembers(TestClassGenerationContext gen
+                {
+                    Envelope.Create(featureSource),
+                    Envelope.Create(featureGherkinDocumentMessage)
+                };
+                envelopes.AddRange(featurePickleMessages.Select(p => Envelope.Create(p)));
 
-                // generate a CodeDom expression to create the Source object from the featureSourceMessage
-                var sourceExpression = new CodeObjectCreateExpression(new CodeTypeReference(typeof(Io.Cucumber.Messages.Types.Source), CodeTypeReferenceOptions.GlobalReference),
-                    new CodePrimitiveExpression(featureSource.Uri),
-                    new CodePrimitiveExpression(featureSource.Data),
-                    new CodeFieldReferenceExpression(new CodeTypeReferenceExpression(new CodeTypeReference(typeof(Io.Cucumber.Messages.Types.SourceMediaType), CodeTypeReferenceOptions.GlobalReference)), featureSource.MediaType.ToString()));
-
-                // generate a CodeDom expression to create the GherkinDocument object from the featureGherkinDocumentMessage
-                var gherkinDocumentExpressionGenerator = new CucumberGherkinDocumentExpressionGenerator();
-                var gherkinDocumentExpression = gherkinDocumentExpressionGenerator.GenerateGherkinDocumentExpression(featureGherkinDocumentMessage);
-
-                // generate a CodeDom expression to create the Pickles object from the featurePickleMessages
-                var pickleExpressionGenerator = new CucumberPicklesExpressionGenerator();
-                var picklesExpression = pickleExpressionGenerator.GeneratePicklesExpression(featurePickleMessages);
-
-                // wrap these expressions in Func<T>
-
-                sourceReturnStatement = new CodeMethodReturnStatement(sourceExpression);
-
-                gherkinDocumentReturnStatement = new CodeMethodReturnStatement(gherkinDocumentExpression);
+                // Serialize each envelope individually
+                ndjsonLines = envelopes
+                    .Select(NdjsonSerializer.Serialize)
+                    .ToArray();
 
-                picklesReturnStatement = new CodeMethodReturnStatement(picklesExpression);
             }
-            catch (Exception e)
+            catch (System.Exception e)
             {
                 generationContext.GenerationWarnings.Add($"WARNING: Failed to process Cucumber Pickles. Support for generating Cucumber Messages will be disabled. Exception: {e.Message}");
                 // Should any error occur during pickling or serialization of Cucumber Messages, we will abort and not add the Cucumber Messages to the featureInfo.
                 // This effectively turns OFF the Cucumber Messages support for this feature.
+                featureFileName = null;
             }
             finally
             {
-                sourceFactoryMethod.Statements.Add(sourceReturnStatement);
+                _featureNdJsonMessages = String.Join(Environment.NewLine, ndjsonLines);
```

Comment:
Why is this in the "finally" block? I think if there is an exception, we don't use it anyway.
Also, I *think* collecting the messages in a StringBuilder is more efficient than collecting them in a string array and only after building up a string from them, but maybe this is just a micro optimization.